dace/codegen/targets/common.py

```diff
@@ -39,8 +39,8 @@ def sym2cpp(s, arrayexprs: Optional[Set[str]] = None) -> Union[str, List[str]]:
     :return: C++-compilable expression or list thereof.
     """
     if not isinstance(s, list):
-        return symbolic.symstr(s, arrayexprs)
-    return [symbolic.symstr(d, arrayexprs) for d in s]
+        return cppunparse.pyexpr2cpp(symbolic.symstr(s, arrayexprs))
```

Comment:
AFAIK `symbolic.symstr` is still returning valid C++ code. Therefore, it is very likely that passing the output of this method to `cppunparse.pyexpr2cpp` will fail. I will add a test.